submissions/webdevtoday/oop_exercise/index.js

```diff
@@ -0,0 +1,80 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Mammal {
+   constructor(name, gender, saying, species, friends) {
+      this.name = name;
+      this.gender = gender;
+      this.species = species;
+      this.saying = saying;
+      this.friends = friends;
+      this.legs = 4;
+   }
+   getProperties() {
+      return [this.species, this.name, this.gender, this.legs, this.friends.join(', ')];
+   }
+   [Symbol.toPrimitive](hint) {
+      if (hint == 'string') {  
+         return this.getProperties().join(';');
+      }
+   }
+}
+
+class Human extends Mammal {
+   constructor(name, gender, saying, friends) {
+      super(name, gender, saying, 'human', friends);
+      this.legs = 2;
+      this.hands = 2;
+   }
+
+   getProperties() {
+      return [this.species, this.name, this.gender, this.legs, this.hands, this.friends.join(', ')];
+   }
```

Comment:
Benefit from a method overload technique.

A child class `methodA` can call an inherited (predecessor's) method via `super.methodA` and supplement it.

This way both parent and child manage only those properties they own directly.

So, get props from the parent by calling `super.getProps`, supplement the list with props unique to a human and return combined array.

Reply:
> Benefit from a method overload technique.
> 
> A child class `methodA` can call an inherited (predecessor's) method via `super.methodA` and supplement it.
> 
> This way both parent and child manage only those properties they own directly.
> 
> So, get props from the parent by calling `super.getProps`, supplement the list with props unique to a human and return combined array.

Yes, I think about it, but I found info about it) I knew only super()